### PR TITLE
SPA Angular 11->12 updates

### LINKED ISSUE #33076: [SPA] Upgrade to angular 12
Here is a list of the changes between a new angular 11 template and a new angular 12 template:

```diff
diff --git "a/.\\angular11\\angularapp\\/README.md" "b/.\\angular12\\angularapp\\/README.md"
index 34f8c3b..89a92ca 100644
--- "a/.\\angular11\\angularapp\\/README.md"
+++ "b/.\\angular12\\angularapp\\/README.md"
@@ -1,6 +1,6 @@
 # Angularapp
 
-This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 11.2.13.
+This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 12.0.2.
 
 ## Development server
 
@@ -12,7 +12,7 @@ Run `ng generate component component-name` to generate a new component. You can
 
 ## Build
 
-Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.
+Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory.
 
 ## Running unit tests
 
@@ -20,7 +20,7 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
 
 ## Running end-to-end tests
 
-Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
+Run `ng e2e` to execute the end-to-end tests via a platform of your choice. To use this command, you need to first add a package that implements end-to-end testing capabilities.
 
 ## Further help
 
diff --git "a/.\\angular11\\angularapp\\/angular.json" "b/.\\angular12\\angularapp\\/angular.json"
index 39db22e..bb0d266 100644
--- "a/.\\angular11\\angularapp\\/angular.json"
+++ "b/.\\angular12\\angularapp\\/angular.json"
@@ -5,7 +5,11 @@
   "projects": {
     "angularapp": {
       "projectType": "application",
-      "schematics": {},
+      "schematics": {
+        "@schematics/angular:application": {
+          "strict": true
+        }
+      },
       "root": "",
       "sourceRoot": "src",
       "prefix": "app",
@@ -18,7 +22,6 @@
             "main": "src/main.ts",
             "polyfills": "src/polyfills.ts",
             "tsConfig": "tsconfig.app.json",
-            "aot": true,
             "assets": [
               "src/favicon.ico",
               "src/assets"
@@ -30,44 +33,48 @@
           },
           "configurations": {
             "production": {
-              "fileReplacements": [
-                {
-                  "replace": "src/environments/environment.ts",
-                  "with": "src/environments/environment.prod.ts"
-                }
-              ],
-              "optimization": true,
-              "outputHashing": "all",
-              "sourceMap": false,
-              "namedChunks": false,
-              "extractLicenses": true,
-              "vendorChunk": false,
-              "buildOptimizer": true,
               "budgets": [
                 {
                   "type": "initial",
-                  "maximumWarning": "2mb",
-                  "maximumError": "5mb"
+                  "maximumWarning": "500kb",
+                  "maximumError": "1mb"
                 },
                 {
                   "type": "anyComponentStyle",
-                  "maximumWarning": "6kb",
-                  "maximumError": "10kb"
+                  "maximumWarning": "2kb",
+                  "maximumError": "4kb"
+                }
+              ],
+              "fileReplacements": [
+                {
+                  "replace": "src/environments/environment.ts",
+                  "with": "src/environments/environment.prod.ts"
                 }
-              ]
+              ],
+              "outputHashing": "all"
+            },
+            "development": {
+              "buildOptimizer": false,
+              "optimization": false,
+              "vendorChunk": true,
+              "extractLicenses": false,
+              "sourceMap": true,
+              "namedChunks": true
             }
-          }
+          },
+          "defaultConfiguration": "production"
         },
         "serve": {
           "builder": "@angular-devkit/build-angular:dev-server",
-          "options": {
-            "browserTarget": "angularapp:build"
-          },
           "configurations": {
             "production": {
               "browserTarget": "angularapp:build:production"
+            },
+            "development": {
+              "browserTarget": "angularapp:build:development"
             }
-          }
+          },
+          "defaultConfiguration": "development"
         },
         "extract-i18n": {
           "builder": "@angular-devkit/build-angular:extract-i18n",
@@ -91,31 +98,6 @@
             ],
             "scripts": []
           }
-        },
-        "lint": {
-          "builder": "@angular-devkit/build-angular:tslint",
-          "options": {
-            "tsConfig": [
-              "tsconfig.app.json",
-              "tsconfig.spec.json",
-              "e2e/tsconfig.json"
-            ],
-            "exclude": [
-              "**/node_modules/**"
-            ]
-          }
-        },
-        "e2e": {
-          "builder": "@angular-devkit/build-angular:protractor",
-          "options": {
-            "protractorConfig": "e2e/protractor.conf.js",
-            "devServerTarget": "angularapp:serve"
-          },
-          "configurations": {
-            "production": {
-              "devServerTarget": "angularapp:serve:production"
-            }
-          }
         }
       }
     }
diff --git "a/.\\angular11\\angularapp\\/e2e/protractor.conf.js" "b/.\\angular11\\angularapp\\/e2e/protractor.conf.js"
deleted file mode 100644
index 361e7f0..0000000
--- "a/.\\angular11\\angularapp\\/e2e/protractor.conf.js"
+++ /dev/null
@@ -1,37 +0,0 @@
-// @ts-check
-// Protractor configuration file, see link for more information
-// https://github.com/angular/protractor/blob/master/lib/config.ts
-
-const { SpecReporter, StacktraceOption } = require('jasmine-spec-reporter');
-
-/**
- * @type { import("protractor").Config }
- */
-exports.config = {
-  allScriptsTimeout: 11000,
-  specs: [
-    './src/**/*.e2e-spec.ts'
-  ],
-  capabilities: {
-    browserName: 'chrome'
-  },
-  directConnect: true,
-  SELENIUM_PROMISE_MANAGER: false,
-  baseUrl: 'http://localhost:4200/',
-  framework: 'jasmine',
-  jasmineNodeOpts: {
-    showColors: true,
-    defaultTimeoutInterval: 30000,
-    print: function() {}
-  },
-  onPrepare() {
-    require('ts-node').register({
-      project: require('path').join(__dirname, './tsconfig.json')
-    });
-    jasmine.getEnv().addReporter(new SpecReporter({
-      spec: {
-        displayStacktrace: StacktraceOption.PRETTY
-      }
-    }));
-  }
-};
\ No newline at end of file
diff --git "a/.\\angular11\\angularapp\\/e2e/src/app.e2e-spec.ts" "b/.\\angular11\\angularapp\\/e2e/src/app.e2e-spec.ts"
deleted file mode 100644
index b00de5e..0000000
--- "a/.\\angular11\\angularapp\\/e2e/src/app.e2e-spec.ts"
+++ /dev/null
@@ -1,23 +0,0 @@
-import { browser, logging } from 'protractor';
-import { AppPage } from './app.po';
-
-describe('workspace-project App', () => {
-  let page: AppPage;
-
-  beforeEach(() => {
-    page = new AppPage();
-  });
-
-  it('should display welcome message', async () => {
-    await page.navigateTo();
-    expect(await page.getTitleText()).toEqual('angularapp app is running!');
-  });
-
-  afterEach(async () => {
-    // Assert that there are no errors emitted from the browser
-    const logs = await browser.manage().logs().get(logging.Type.BROWSER);
-    expect(logs).not.toContain(jasmine.objectContaining({
-      level: logging.Level.SEVERE,
-    } as logging.Entry));
-  });
-});
diff --git "a/.\\angular11\\angularapp\\/e2e/src/app.po.ts" "b/.\\angular11\\angularapp\\/e2e/src/app.po.ts"
deleted file mode 100644
index c9c85ab..0000000
--- "a/.\\angular11\\angularapp\\/e2e/src/app.po.ts"
+++ /dev/null
@@ -1,11 +0,0 @@
-import { browser, by, element } from 'protractor';
-
-export class AppPage {
-  async navigateTo(): Promise<unknown> {
-    return browser.get(browser.baseUrl);
-  }
-
-  async getTitleText(): Promise<string> {
-    return element(by.css('app-root .content span')).getText();
-  }
-}
diff --git "a/.\\angular11\\angularapp\\/e2e/tsconfig.json" "b/.\\angular11\\angularapp\\/e2e/tsconfig.json"
deleted file mode 100644
index 0782539..0000000
--- "a/.\\angular11\\angularapp\\/e2e/tsconfig.json"
+++ /dev/null
@@ -1,13 +0,0 @@
-/* To learn more about this file see: https://angular.io/config/tsconfig. */
-{
-  "extends": "../tsconfig.json",
-  "compilerOptions": {
-    "outDir": "../out-tsc/e2e",
-    "module": "commonjs",
-    "target": "es2018",
-    "types": [
-      "jasmine",
-      "node"
-    ]
-  }
-}
diff --git "a/.\\angular11\\angularapp\\/package.json" "b/.\\angular12\\angularapp\\/package.json"
index 7e18226..6847845 100644
--- "a/.\\angular11\\angularapp\\/package.json"
+++ "b/.\\angular12\\angularapp\\/package.json"
@@ -5,41 +5,35 @@
     "ng": "ng",
     "start": "ng serve",
     "build": "ng build",
-    "test": "ng test",
-    "lint": "ng lint",
-    "e2e": "ng e2e"
+    "watch": "ng build --watch --configuration development",
+    "test": "ng test"
   },
   "private": true,
   "dependencies": {
-    "@angular/animations": "~11.2.14",
-    "@angular/common": "~11.2.14",
-    "@angular/compiler": "~11.2.14",
-    "@angular/core": "~11.2.14",
-    "@angular/forms": "~11.2.14",
-    "@angular/platform-browser": "~11.2.14",
-    "@angular/platform-browser-dynamic": "~11.2.14",
-    "@angular/router": "~11.2.14",
+    "@angular/animations": "~12.0.2",
+    "@angular/common": "~12.0.2",
+    "@angular/compiler": "~12.0.2",
+    "@angular/core": "~12.0.2",
+    "@angular/forms": "~12.0.2",
+    "@angular/platform-browser": "~12.0.2",
+    "@angular/platform-browser-dynamic": "~12.0.2",
+    "@angular/router": "~12.0.2",
     "rxjs": "~6.6.0",
-    "tslib": "^2.0.0",
-    "zone.js": "~0.11.3"
+    "tslib": "^2.1.0",
+    "zone.js": "~0.11.4"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.1102.13",
-    "@angular/cli": "~11.2.13",
-    "@angular/compiler-cli": "~11.2.14",
+    "@angular-devkit/build-angular": "~12.0.2",
+    "@angular/cli": "~12.0.2",
+    "@angular/compiler-cli": "~12.0.2",
     "@types/jasmine": "~3.6.0",
     "@types/node": "^12.11.1",
-    "codelyzer": "^6.0.0",
-    "jasmine-core": "~3.6.0",
-    "jasmine-spec-reporter": "~5.0.0",
-    "karma": "~6.1.0",
+    "jasmine-core": "~3.7.0",
+    "karma": "~6.3.0",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coverage": "~2.0.3",
     "karma-jasmine": "~4.0.0",
-    "karma-jasmine-html-reporter": "~1.5.0",
-    "protractor": "~7.0.0",
-    "ts-node": "~8.3.0",
-    "tslint": "~6.1.0",
-    "typescript": "~4.1.5"
+    "karma-jasmine-html-reporter": "^1.5.0",
+    "typescript": "~4.2.3"
   }
 }
diff --git "a/.\\angular11\\angularapp\\/src/app/app.component.html" "b/.\\angular12\\angularapp\\/src/app/app.component.html"
index 12938ab..a751c67 100644
--- "a/.\\angular11\\angularapp\\/src/app/app.component.html"
+++ "b/.\\angular12\\angularapp\\/src/app/app.component.html"
@@ -433,7 +433,7 @@
       <pre *ngSwitchCase="'pwa'">ng add @angular/pwa</pre>
       <pre *ngSwitchCase="'dependency'">ng add _____</pre>
       <pre *ngSwitchCase="'test'">ng test</pre>
-      <pre *ngSwitchCase="'build'">ng build --prod</pre>
+      <pre *ngSwitchCase="'build'">ng build</pre>
   </div>
 
   <!-- Links -->
@@ -461,17 +461,6 @@
       </svg>
     </a>
 
-    <a class="circle-link" title="Protractor" href="https://www.protractortest.org/" target="_blank" rel="noopener">
-      <svg xmlns="http://www.w3.org/2000/svg" width="21.81" height="23.447" viewBox="0 0 21.81 23.447">
-        <title>Angular Protractor Logo</title>
-        <g id="Group_26" data-name="Group 26" transform="translate(0)">
-          <path id="Path_28" data-name="Path 28" d="M4620.155,311.417h0l-10.881,3.9,1.637,14.4,9.244,5.152,9.244-5.152,1.685-14.4Z" transform="translate(-4609.274 -311.417)" fill="#e13439"/>
-          <path id="Path_29" data-name="Path 29" d="M4702.088,311.417v23.447l9.244-5.152,1.685-14.4Z" transform="translate(-4691.207 -311.417)" fill="#b52f32"/>
-          <path id="Path_30" data-name="Path 30" d="M4651.044,369.58v-.421h1.483a7.6,7.6,0,0,0-2.106-5.052l-1.123,1.123-.3-.3,1.122-1.121a7.588,7.588,0,0,0-4.946-2.055v1.482h-.421v-1.485a7.589,7.589,0,0,0-5.051,2.058l1.122,1.121-.3.3-1.123-1.123a7.591,7.591,0,0,0-2.106,5.052h1.482v.421h-1.489v1.734h15.241V369.58Zm-10.966-.263a4.835,4.835,0,0,1,9.67,0Z" transform="translate(-4634.008 -355.852)" fill="#fff"/>
-        </g>
-      </svg>
-    </a>
-
     <a class="circle-link" title="Find a Local Meetup" href="https://www.meetup.com/find/?keywords=angular" target="_blank" rel="noopener">
       <svg xmlns="http://www.w3.org/2000/svg" width="24.607" height="23.447" viewBox="0 0 24.607 23.447">
         <title>Meetup Logo</title>
diff --git "a/.\\angular11\\angularapp\\/src/environments/environment.ts" "b/.\\angular12\\angularapp\\/src/environments/environment.ts"
index 7b4f817..f56ff47 100644
--- "a/.\\angular11\\angularapp\\/src/environments/environment.ts"
+++ "b/.\\angular12\\angularapp\\/src/environments/environment.ts"
@@ -1,5 +1,5 @@
 // This file can be replaced during build by using the `fileReplacements` array.
-// `ng build --prod` replaces `environment.ts` with `environment.prod.ts`.
+// `ng build` replaces `environment.ts` with `environment.prod.ts`.
 // The list of file replacements can be found in `angular.json`.
 
 export const environment = {
@@ -13,4 +13,4 @@ export const environment = {
  * This import should be commented out in production mode because it will have a negative impact
  * on performance if an error is thrown.
  */
-// import 'zone.js/dist/zone-error';  // Included with Angular CLI.
+// import 'zone.js/plugins/zone-error';  // Included with Angular CLI.
diff --git "a/.\\angular11\\angularapp\\/src/polyfills.ts" "b/.\\angular12\\angularapp\\/src/polyfills.ts"
index d5f67bd..373f538 100644
--- "a/.\\angular11\\angularapp\\/src/polyfills.ts"
+++ "b/.\\angular12\\angularapp\\/src/polyfills.ts"
@@ -57,7 +57,7 @@
 /***************************************************************************************************
  * Zone JS is required by default for Angular itself.
  */
-import 'zone.js/dist/zone';  // Included with Angular CLI.
+import 'zone.js';  // Included with Angular CLI.
 
 
 /***************************************************************************************************
diff --git "a/.\\angular11\\angularapp\\/src/test.ts" "b/.\\angular12\\angularapp\\/src/test.ts"
index 50193eb..2042356 100644
--- "a/.\\angular11\\angularapp\\/src/test.ts"
+++ "b/.\\angular12\\angularapp\\/src/test.ts"
@@ -1,6 +1,6 @@
 // This file is required by karma.conf.js and loads recursively all the .spec and framework files
 
-import 'zone.js/dist/zone-testing';
+import 'zone.js/testing';
 import { getTestBed } from '@angular/core/testing';
 import {
   BrowserDynamicTestingModule,
diff --git "a/.\\angular11\\angularapp\\/tsconfig.json" "b/.\\angular12\\angularapp\\/tsconfig.json"
index 4a4dc62..6df8283 100644
--- "a/.\\angular11\\angularapp\\/tsconfig.json"
+++ "b/.\\angular12\\angularapp\\/tsconfig.json"
@@ -4,13 +4,17 @@
   "compilerOptions": {
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
     "sourceMap": true,
     "declaration": false,
     "downlevelIteration": true,
     "experimentalDecorators": true,
     "moduleResolution": "node",
     "importHelpers": true,
-    "target": "es2015",
+    "target": "es2017",
     "module": "es2020",
     "lib": [
       "es2018",
@@ -18,6 +22,9 @@
     ]
   },
   "angularCompilerOptions": {
-    "enableI18nLegacyMessageIdFormat": false
+    "enableI18nLegacyMessageIdFormat": false,
+    "strictInjectionParameters": true,
+    "strictInputAccessModifiers": true,
+    "strictTemplates": true
   }
 }
diff --git "a/.\\angular11\\angularapp\\/tslint.json" "b/.\\angular11\\angularapp\\/tslint.json"
deleted file mode 100644
index 277c8eb..0000000
--- "a/.\\angular11\\angularapp\\/tslint.json"
+++ /dev/null
@@ -1,152 +0,0 @@
-{
-  "extends": "tslint:recommended",
-  "rulesDirectory": [
-    "codelyzer"
-  ],
-  "rules": {
-    "align": {
-      "options": [
-        "parameters",
-        "statements"
-      ]
-    },
-    "array-type": false,
-    "arrow-return-shorthand": true,
-    "curly": true,
-    "deprecation": {
-      "severity": "warning"
-    },
-    "eofline": true,
-    "import-blacklist": [
-      true,
-      "rxjs/Rx"
-    ],
-    "import-spacing": true,
-    "indent": {
-      "options": [
-        "spaces"
-      ]
-    },
-    "max-classes-per-file": false,
-    "max-line-length": [
-      true,
-      140
-    ],
-    "member-ordering": [
-      true,
-      {
-        "order": [
-          "static-field",
-          "instance-field",
-          "static-method",
-          "instance-method"
-        ]
-      }
-    ],
-    "no-console": [
-      true,
-      "debug",
-      "info",
-      "time",
-      "timeEnd",
-      "trace"
-    ],
-    "no-empty": false,
-    "no-inferrable-types": [
-      true,
-      "ignore-params"
-    ],
-    "no-non-null-assertion": true,
-    "no-redundant-jsdoc": true,
-    "no-switch-case-fall-through": true,
-    "no-var-requires": false,
-    "object-literal-key-quotes": [
-      true,
-      "as-needed"
-    ],
-    "quotemark": [
-      true,
-      "single"
-    ],
-    "semicolon": {
-      "options": [
-        "always"
-      ]
-    },
-    "space-before-function-paren": {
-      "options": {
-        "anonymous": "never",
-        "asyncArrow": "always",
-        "constructor": "never",
-        "method": "never",
-        "named": "never"
-      }
-    },
-    "typedef": [
-      true,
-      "call-signature"
-    ],
-    "typedef-whitespace": {
-      "options": [
-        {
-          "call-signature": "nospace",
-          "index-signature": "nospace",
-          "parameter": "nospace",
-          "property-declaration": "nospace",
-          "variable-declaration": "nospace"
-        },
-        {
-          "call-signature": "onespace",
-          "index-signature": "onespace",
-          "parameter": "onespace",
-          "property-declaration": "onespace",
-          "variable-declaration": "onespace"
-        }
-      ]
-    },
-    "variable-name": {
-      "options": [
-        "ban-keywords",
-        "check-format",
-        "allow-pascal-case"
-      ]
-    },
-    "whitespace": {
-      "options": [
-        "check-branch",
-        "check-decl",
-        "check-operator",
-        "check-separator",
-        "check-type",
-        "check-typecast"
-      ]
-    },
-    "component-class-suffix": true,
-    "contextual-lifecycle": true,
-    "directive-class-suffix": true,
-    "no-conflicting-lifecycle": true,
-    "no-host-metadata-property": true,
-    "no-input-rename": true,
-    "no-inputs-metadata-property": true,
-    "no-output-native": true,
-    "no-output-on-prefix": true,
-    "no-output-rename": true,
-    "no-outputs-metadata-property": true,
-    "template-banana-in-box": true,
-    "template-no-negated-async": true,
-    "use-lifecycle-interface": true,
-    "use-pipe-transform-interface": true,
-    "directive-selector": [
-      true,
-      "attribute",
-      "app",
-      "camelCase"
-    ],
-    "component-selector": [
-      true,
-      "element",
-      "app",
-      "kebab-case"
-    ]
-  }
-}
```

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/dotnet/aspnetcore/issues/33076

Note: When individual auth is enabled, ng test run on the ClientApp will fail due to issues with webpack 5 removing polyfills for node.js core modules, specifically util, and then process due to bootstrap.  

ng test output errors
```
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }
```

But other wise, the angular template seems to runs fine with this PR

### CODE REVIEW

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/angular.json [hunk 1]
```diff
@@ -30,35 +34,50 @@
...
           "builder": "@angular-devkit/build-angular:dev-server",
-          "options": {
-            "browserTarget": "Company.WebApplication1:build",
-            "proxyConfig": "proxy.conf.js"
-          },
           "configurations": {
             "production": {
-              "browserTarget": "Company.WebApplication1:build:production"
+              "browserTarget": "Company.WebApplication1:build:production",
+              "proxyConfig": "proxy.conf.js"
```
Comment (javiercn) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/angular.json:73]: I don't think this is needed? We shouldn't be using any proxying on the prod environment.

**Comment by javiercn** [comment]
It's fine if we do this, I remember we hit a similar thing when we upgraded to webpack 5 on SignalR and other places.

```console
install 'util'
add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
```

**Comment by HaoK** [comment]
It wasn't clear how best I could do this easily since angular doesn't expose webpack directly, have to do some kind of webpack custom builder?

**Comment by javiercn** [comment]
@HaoK ahh, that's true.

How does this work on a new template?

**Comment by HaoK** [comment]
I'm not sure what codepaths in the auth that actually rely on util or process, its the ng test command that runs into this.  I was able to login normally just fine, so I'm not sure if there are actual functionality that stops working due to util/process/browser changes.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/angular.json (hunk 1 above)
Comment (HaoK) [reply src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/angular.json]: Deleting this is good, it was calling require process which is one of the polyfill things

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/proxy.conf.js [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-      "/connect",
-      "/ApplyDatabaseMigrations",
-//#endif
-   ],
-    target: target,
-    secure: false
-  }
-]
-
-module.exports = PROXY_CONFIG;
```
Comment (javiercn) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/proxy.conf.js:23]: Removing this file is wrong. This is needed to proxy to the backend

**Comment by javiercn** [comment]
@HaoK can you point out what files are causing grief to the tests?

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/proxy.conf.js (hunk 2 above)
Comment (HaoK) [reply src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/proxy.conf.js:23]: I misinterpreted your comment about this in angular.json then, so just delete the entry in angular.json and leave the file?

Comment (javiercn) [reply src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/proxy.conf.js:23]: Yep, only the entry on the production bit. On development we still need the entry, and we need the file.

**Comment by HaoK** [comment]
@javiercn the login/logout files:

```

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }

./src/api-authorization/logout/logout.component.spec.ts:6:0-27 - Error: Module not found: Error: Can't resolve 'util' in 'C:\angular12\ClientApp\src\api-authorization\logout'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }


√ Browser application bundle generation complete.

./src/api-authorization/login/login.component.spec.ts:7:0-27 - Error: Module not found: Error: Can't resolve 'util' in 'C:\angular12\ClientApp\src\api-authorization\login'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }

./src/api-authorization/logout/logout.component.spec.ts:6:0-27 - Error: Module not found: Error: Can't resolve 'util' in 'C:\angular12\ClientApp\src\api-authorization\logout'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }
```

**Comment by HaoK** [comment]
So I can make the errors go away and the tests pass if I just delete the log/util in both of the spec files, maybe that's the easiest fix

**Comment by javiercn** [comment]
> So I can make the errors go away and the tests pass if I just delete the logging/util usage in both of the spec files, maybe that's the easiest fix

That sounds like an appropriate fix.

**Comment by HaoK** [comment]
Cool things look good now with ng test, I think this PR is ready to go

**Review by javiercn — APPROVED** [review]

**Pull request merged**